spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/store/MongoDbChannelMessageStore.java

```diff
@@ -96,6 +97,14 @@ public boolean isPriorityEnabled() {
 	@Override
 	public void afterPropertiesSet() {
 		super.afterPropertiesSet();
+		if (this.getCreateIndex()) {
```

Comment:
This check and new method call has to be done in a super class. Probably this `afterPropertiesSet()` can be removed from here altogether then.

Reply:
In the previous implement `MongoDbChannelMessageStore` subclass creates an index as well. Here is the index creates by the subclass `MongoDbChannelMessageStore`: `getMongoTemplate()
				.indexOps(this.collectionName)
				.ensureIndex(new Index(MessageDocumentFields.GROUP_ID, Sort.Direction.ASC)
						.on(MessageDocumentFields.PRIORITY, Sort.Direction.DESC)
						.on(MessageDocumentFields.LAST_MODIFIED_TIME, Sort.Direction.ASC)
						.on(MessageDocumentFields.SEQUENCE, Sort.Direction.ASC));`

So, I suppose this class must create an index after the super class created its indexes. It's Why I overrode the `createIndexes()` in the subclass and check in that `afterPropertiesSet()` which will call the `createIndexes()` in the subclass. The `createIndex()` will then call its super `createIndex()`. You can see the implementation of `createIndexes()` in `MongoDbChannelMessageStore`.

If I remove `afterPropertiesSet()` from subclass, I don't know how we will create the subclass `MongoDbChannelMessageStore` index.

Reply:
That is called `polymorphism` in Java: when super class calls the method, the actual instance is called. So, if we override `createIndexes()` in the sub-class then exactly this one is called. Therefore no need to call it from sub-class if we call it from super class.
And since we move all the logic to that `createIndexes()` extension, there is nothing left in its `afterPropertiesSet()`, therefore we don't need it any more.